### PR TITLE
Add workflow field to `ClientError`, `ClientTimeout`

### PR DESCRIPTION
**Pull request by MetRonnie**
Needed for https://github.com/cylc/cylc-uiserver/pull/379

**Check List**

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Tests are not needed.
- [x] No changelog entry required
- [x] No docs update needed.

### CODE REVIEW

**Review by wxtim — APPROVED** [review]
Yep

**Comment by oliver-sanders** [comment]
The UIS PR is up against 1.2.0 which will use `8.1.0` so this should go against `8.1.0` too.

**Comment by MetRonnie** [comment]
It depends on if we want to do a maintenance release for the UIS? (No milestone for that currently)

**Comment by oliver-sanders** [comment]
I don't think we need a UIS maintenance release for this.

File: cylc/flow/exceptions.py [hunk 1]
```diff
@@ -254,9 +254,17 @@ def __str__(self):
 
 class ClientError(CylcError):
 
-    def __init__(self, message: str, traceback: Optional[str] = None):
+    def __init__(
+        self,
+        message: str,
+        traceback: Optional[str] = None,
+        workflow: Optional[str] = None
```
Comment (oliver-sanders) [inline cylc/flow/exceptions.py:261]: Might be a good idea to call this `workflow_id` as we also have the tokenised variant.

**Review by oliver-sanders — COMMENTED** [review]
LGTM, gotta couple of unexpected commits here.

**Comment by MetRonnie** [comment]
> LGTM, gotta couple of unexpected commits here.

Ah, that would be because the base was changed from 8.0.x to master, so it wouldn't have mattered (I'll rebase anyway)

File: cylc/flow/exceptions.py (hunk 1 above)
Comment (MetRonnie) [reply cylc/flow/exceptions.py:261]: Don't think it matters much for these exceptions. The more important one would be `WorkflowRuntimeClient`
https://github.com/cylc/cylc-flow/blob/af64ace2f078b7d656c83c09db483a531f43b98f/cylc/flow/network/client.py#L64-L66

however that's probably out of scope

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**